wal/wal.go

```diff
@@ -223,6 +230,30 @@ func Create(lg *zap.Logger, dirpath string, metadata []byte) (*WAL, error) {
 	return w, nil
 }
 
+func (w *WAL) cleanupWAL(lg *zap.Logger) {
+	var err error
+	if err = w.Close(); err != nil {
+		if lg != nil {
+			lg.Panic("failed to cleanup WAL", zap.Error(err))
```

Comment:
About the log message here, for consistency and readability, would suggest something like `failed to close WAL during cleanup` similar to what you have on LOC 246.